### PR TITLE
Support for pod labels.

### PR DESCRIPTION
**Pull request by jmencak**

### CODE REVIEW

File: cmd/openshift-tuned.go [hunk 1]
```diff
@@ -164,51 +202,58 @@ func nodeLabelsRead() map[string]string {
...
 		tunedReload()
 		return false
 	}
-	for key, value := range nodeLabelsNew {
-		if nodeLabelsOld[key] != value {
-			log.Printf("%s: node label[%s] == %s (old value: %s)\n", PNAME, key, value, nodeLabelsOld[key])
-			nodeLabelsDump(nodeLabelsNew)
+	for key, value := range labelsNew {
+		// No need to sort labelsOld[key] and value, already sorted when reading/adding label values
+		if !reflect.DeepEqual(labelsOld[key], value) {
```
Comment (sjug) [inline cmd/openshift-tuned.go:248]: Using `DeepEqual` to compare arrays?

File: cmd/openshift-tuned.go [hunk 2]
```diff
@@ -128,33 +146,53 @@ func tunedReload() {
...
+	}
+	if err != nil {
 		panic(err.Error())
 	}
 
-	return node.Labels
+	// This is not strictly necessary, but be consistent with podLabelsGet and return the same type
+	nodeLabels := map[string][]string{}
+	for key, value := range node.Labels {
+		nodeLabels[key] = append(nodeLabels[key], value)
```
Comment (sjug) [inline cmd/openshift-tuned.go:163]: Why do the node labels just need `append()` vs the `insertUnique()` for the pod labels below?

File: cmd/openshift-tuned.go [hunk 3]
```diff
@@ -253,9 +299,15 @@ func mainLoop(clientset *kubernetes.Clientset, nodeName string, resyncPeriodDura
 	ticker := time.NewTicker(time.Second * time.Duration(resyncPeriodDuration))
 	go func() {
 		for range ticker.C {
+			// Compare node labels
 			nodeLabels := nodeLabelsGet(clientset, nodeName)
-			nodeLabelsCompare(nodeLabelsOld, nodeLabels)
+			labelsCompare(nodeLabelsOld, nodeLabels, "node", fileNodeLabels)
```
Comment (sjug) [inline cmd/openshift-tuned.go:304]: If `fileNodeLabels` is a global variable you don't need to pass it around, maybe it doesn't need to be global?

File: cmd/openshift-tuned.go [hunk 4]
```diff
@@ -164,51 +202,58 @@ func nodeLabelsRead() map[string]string {
...
+			}
 		}
 	}
 	f.Sync()
 }
 
-func nodeLabelsCompare(nodeLabelsOld map[string]string, nodeLabelsNew map[string]string) bool {
-	if nodeLabelsOld == nil {
-		/* no node labels defined yet */
+func labelsCompare(labelsOld map[string][]string, labelsNew map[string][]string, labelType string, fileLabels string) bool {
```
Comment (sjug) [inline cmd/openshift-tuned.go:235]: Consider having a struct for labels that has a labelType field (and one field with the label map itself).

**Review by sjug — COMMENTED** [review]
Few thoughts

File: cmd/openshift-tuned.go (hunk 1 above)
Comment (jmencak) [reply cmd/openshift-tuned.go:248]: Why not?

File: cmd/openshift-tuned.go (hunk 2 above)
Comment (jmencak) [reply cmd/openshift-tuned.go:163]: Aren't node labels for a single node always unique?

File: cmd/openshift-tuned.go (hunk 3 above)
Comment (jmencak) [reply cmd/openshift-tuned.go:304]: Possibly.

File: cmd/openshift-tuned.go (hunk 4 above)
Comment (jmencak) [reply cmd/openshift-tuned.go:235]: Possibly.

**Comment by jmencak** [comment]
/hold

Thanks for the review!  Will restructure the whole PR, and perhaps even cancel as I'll be adding node/pod watch support.

File: cmd/openshift-tuned.go (hunk 1 above)
Comment (sjug) [reply cmd/openshift-tuned.go:248]: Seems unnecessary

File: cmd/openshift-tuned.go (hunk 2 above)
Comment (sjug) [reply cmd/openshift-tuned.go:163]: Good call

**Comment by jmencak** [comment]
Closing in favour of https://github.com/openshift/openshift-tuned/pull/9

**Pull request closed without merging**